### PR TITLE
[FLAVA] Add checkpoint test

### PR DESCRIPTION
**Pull request by RdoubleA**
Summary:
Unification may alter the architecture of some of the models in the codebase. A model checkpoint loading tests is necessary as a quick sanity check for any major layer changes. This PR adds such a test for FLAVA. 

Test plan:
`pytest test/models/flava/test_flava_checkpoint.py -vv`

```
======================================================== test session starts ========================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /fsx/users/rafiayub/conda/envs/torchmm/bin/python
cachedir: .pytest_cache
rootdir: /data/home/rafiayub/torchmultimodal, configfile: pyproject.toml
plugins: cov-3.0.0, mock-3.8.2
collected 2 items                                                                                                                   

test/models/flava/test_flava_checkpoint.py::TestFLAVACheckpoint::test_flava_model_for_classification PASSED                   [ 50%]
test/models/flava/test_flava_checkpoint.py::TestFLAVACheckpoint::test_flava_model_for_pretraining PASSED                      [100%]
```

### CODE REVIEW

File: examples/flava/test/test_flava_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    def test_flava_model_for_pretraining(self, pretraining_model, inputs_pretraining):
+        output = pretraining_model(*inputs_pretraining("mm"))
+        actual = sum(
+            value if value is not None else 0 for value in output.losses.values()
+        )
+        expected = torch.tensor(29.2157)
+        assert_expected(actual, expected, rtol=0, atol=1e-4)
+
+        output = pretraining_model(*inputs_pretraining("image"))
+        actual = sum(
```
Comment (ankitade) [inline examples/flava/test/test_flava_checkpoint.py:103]: lets just test out all 5 loss values for this test.

**Review by ankitade — COMMENTED** [review]
why examples? wouldnt mind putting this in core.

**Comment by RdoubleA** [comment]
> why examples? wouldnt mind putting this in core.

Originally I thought it would be a slow test due to checkpoint loading so I put it in examples but it’s actually pretty quick, so I’ll move it to core

**Review by langong347 — COMMENTED** [review]
Where is the step that loads the checkpoint? Or is it more of a regression test with fixed rng state?

Do you mean to put this under `examples/` or core? In your test plan it says `examples/` but in the change list it says core.

**Comment by RdoubleA** [comment]
> Where is the step that loads the checkpoint? Or is it more of a regression test with fixed rng state?
> 
> Do you mean to put this under `examples/` or core? In your test plan it says `examples/` but in the change list it says core.

@langong347 The `pretrained_model_key` argument triggers loading the checkpoint, see [here](https://github.com/facebookresearch/multimodal/blob/ff50cdec1b4b019fdb0939082a6f942c434a6ddf/torchmultimodal/models/flava/flava_model.py#L466)
